singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java

```diff
@@ -182,7 +196,41 @@ public void onClick(View v) {
         mappedIn = new MappedIn(getApplication());
         mappedIn.getVenues(new getVenuesCallback());
 
-        loading = (TextView) findViewById(R.id.loading_textview);
+        loading = (TextView) findViewById(R.id.venue_loading_textview);
+
+        //search
+        search = (EditText) findViewById(R.id.search_edit_text);
+        suggestListView = (ListView) findViewById(R.id.suggest_list_view);
+        searchGridView = (GridView) findViewById(R.id.search_grid_view);
+
+        search.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if (actionId == EditorInfo.IME_ACTION_SEARCH){
+                    suggestListView.setVisibility(View.INVISIBLE);
+                    smartSearch.search(search.getText().toString(), 1000);
+                    return true;
+                }
+                return false;
+            }
+        });
+        search.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
```

Comment:
if these are no-ops, can we lave them blank?

Reply:
Couldn't do it in Java, it request me Override all method in that interface